apps/web-giddh/src/app/invoice/preview/models/invoice-preview-details/invoice-preview-details.component.ts

```diff
@@ -526,6 +533,49 @@ export class InvoicePreviewDetailsComponent implements OnInit, OnChanges, AfterV
      * @memberof InvoicePreviewDetailsComponent
      */
     public openInvoiceAdvanceReceiptModal(): void {
-        this.onOpenAdvanceReceiptModal.emit(true);
+        if (this.isAccountHaveAdvanceReceipts) {
+            this.onOpenAdvanceReceiptModal.emit(true);
+        }
+    }
+    /**
+     * To toggle change status container
+     *
+     * @param {ReciptResponse} item selected row item data
+     * @memberof InvoicePreviewDetailsComponent
+     */
+    public clickChangeStatusToggle(item: any): void {
+        if (!this.isAccountHaveAdvanceReceipts) {
+            if (item && item.account && item.account.uniqueName && item.voucherDate) {
+                this.getAllAdvanceReceipts(item.account.uniqueName, item.voucherDate);
+            }
+        }
+    }
+
+    /**
+     * Call API to get all advance receipts of an invoice
+     *
+     * @param {*} customerUniquename Selected customer unique name
+     * @param {*} voucherDate Voucher Date (DD-MM-YYYY)
+     * @memberof InvoicePreviewDetailsComponent
+     */
+    public getAllAdvanceReceipts(customerUniqueName: string, voucherDate: string): void {
+        if (customerUniqueName && voucherDate) {
+            let requestObject = {
+                accountUniqueName: customerUniqueName,
+                invoiceDate: voucherDate
+            };
+            this.salesService.getAllAdvanceReceiptVoucher(requestObject).subscribe(res => {
+                if (res && res.status === 'success') {
+                    if (res.body && res.body.length) {
+                        this.isAccountHaveAdvanceReceipts = true;
+                    } else {
+                        this.isAccountHaveAdvanceReceipts = false;
+                    }
+                    this.detectChanges();
+                } else {
+                    this.isAccountHaveAdvanceReceipts = false;
```

Comment:
Add detect changes here also

Reply:
No need to detect here because in case of else there we have nothing to be change

Reply:
?